### PR TITLE
Add floating point comparison warnings and fix usages

### PR DESCRIPTION
**Pull request by EricWF**

### CODE REVIEW

File: test/filter_test.cc [hunk 1]
```diff
@@ -69,7 +69,7 @@ BENCHMARK(BM_CalculatePi)->ThreadPerCpu();
 int main(int argc, const char* argv[]) {
   benchmark::Initialize(&argc, argv);
 
-  assert(CalculatePi(1) == 0.0);
+  assert(std::fabs(CalculatePi(1)) < 0.001);
```
Comment (dominichamon) [inline test/filter_test.cc]: std::numeric_limits<double>::epsilon() ?

Comment (EricWF) [inline test/filter_test.cc]: Didn't know that existed. Thanks.

**Comment by dominichamon** [comment]
LGTM

**Comment by pphaneuf** [comment]
Seems to be a small hiccup with Google interns not being included as Google employees, somehow. You should feel free to ignore this, as long as you're a Google intern. :wink: 

(but I'll try to get it fixed)

**Pull request merged**